nomos-services/data-availability/sampling/src/lib.rs

```diff
@@ -197,20 +242,80 @@ where
         }
     }
 
+    async fn request_commitments(
+        storage_adapter: &DaStorage,
+        api_request: &UnboundedSender<ApiRequest>,
+        blob_id: BlobId,
+    ) -> Option<DaBlobSharedCommitments> {
+        // First try to get from storage which most of the time should be the case
+        if let Ok(Some(commitments)) = storage_adapter.get_commitments(blob_id).await {
+            return Some(commitments);
+        }
+
+        error!("Failed to get commitments from storage adapter");
+
+        // Fall back to API request
+        if let Err(e) = api_request.send(ApiRequest::GetCommitments { blob_id }) {
+            error!("Error sending request to API backend: {e}");
+        }
+
+        None
+    }
+
+    async fn handle_api_response(
+        api_response: ApiResponse,
+        network_adapter: &mut DaNetwork,
+        sampler: &mut Backend,
+    ) {
+        match api_response {
+            ApiResponse::Commitments {
+                blob_id,
+                commitments: Some(commitments),
+            } => {
+                info_with_id!(blob_id, "ReceivedCommitments");
+                if let SamplingState::Init(sampling_subnets) =
+                    sampler.init_sampling(blob_id, commitments).await
```

Comment:
If ApiRequest included the oneshot response channel, then sampling would be triggered in one place.